testsuites/syncgateway/functional/tests/test_cbs_collections.py

```diff
@@ -261,6 +261,48 @@ def test_collection_channels(scopes_collections_tests_fixture):
     e.match("Not Found")
 
 
+@pytest.mark.syncgateway
+@pytest.mark.collections
+def test_restricted_collection(scopes_collections_tests_fixture):
+    """
+    1. Create second collection on CB server
+    2. Add documents to the collections on CB server
+    3. Sync one collection to SGW
+    4. Check that documents that are in the server restricted collection are not accessible via SGW
+    """
+
+    if is_using_views:
+        pytest.skip("""It is not necessary to run scopes and collections tests with views.
+                When it is enabled, there is a problem that affects the rest of the tests suite.""")
+
+    sg_client, sg_admin_url, db, scope, collection = scopes_collections_tests_fixture
+    # 1. Create second collection on CB server
+    random_suffix = str(uuid.uuid4())[:8]
+    second_collection = "collection_" + random_suffix
+    cb_server.create_collection(bucket, scope, second_collection)
+
+    doc_1_key = "doc_1" + random_suffix
+    doc_2_key = "doc_2" + random_suffix
+
+    # 2. Add a document to each collection
+    cb_server.add_simple_document(bucket, scope, collection, doc_1_key)
+    cb_server.add_simple_document(bucket, scope, second_collection, doc_2_key)
+
+    assert(cb_server.get_document(bucket, scope, collection, doc_1_key)["id"] == doc_1_key)
+    assert(cb_server.get_document(bucket, scope, second_collection, doc_2_key)["id"] == doc_2_key)
+
+    # 3. Sync one collection to SGW
+    db_config = {"bucket": bucket, "scopes": {scope: {"collections": {collection: {}}}}, "num_index_replicas": 0,
+                 "import_docs": True, "enable_shared_bucket_access": True}
+    admin_client.post_db_config(db, db_config)
+    admin_client.wait_for_db_online(db, 60)
+
+    # 4. Check that documents in the server restricted collection are not accesible via SGW
+    all_docs = sg_client.get_all_docs(sg_admin_url, db)
+    assert(all_docs["total_rows"] == 1)
```

Comment:
It'll be good to give a description in case the test fails, so it'll be easier from just looking at the logs to understand what's going on. For example: "The number of documents is different from the expected. This could be because restricted files are being picked up when they shouldn't" (you probably have a more concise and informative description than this).

Reply:
some descriptions for test failure added, let me know if they need more detail